lib/location/location.c

```diff
@@ -25,17 +25,70 @@ BUILD_ASSERT(
 	IS_ENABLED(CONFIG_LOCATION_METHOD_WIFI),
 	"At least one location method must be enabled");
 
+#define LOCATION_METHOD_UNKNOWN 0xFF
+#define LOCATION_METHOD_NONE 0xFE
+
 static bool initialized;
 
 static const char LOCATION_METHOD_CELLULAR_STR[] = "Cellular";
 static const char LOCATION_METHOD_GNSS_STR[] = "GNSS";
 static const char LOCATION_METHOD_WIFI_STR[] = "Wi-Fi";
 static const char LOCATION_METHOD_UNKNOWN_STR[] = "Unknown";
 
+enum location_method location_str_to_method(char *method_string)
+{
+	enum location_method method_type = LOCATION_METHOD_UNKNOWN;
+
+	if (strcmp(method_string, "gnss") == 0) {
+		method_type = LOCATION_METHOD_GNSS;
+	} else if (strcmp(method_string, "wifi") == 0) {
+		method_type = LOCATION_METHOD_WIFI;
+	} else if (strcmp(method_string, "cellular") == 0) {
+		method_type = LOCATION_METHOD_CELLULAR;
+	} else if (strcmp(method_string, "") == 0) {
+		method_type = LOCATION_METHOD_NONE;
+	}
+	return method_type;
+}
+
 int location_init(location_event_handler_t handler)
 {
 	int err;
 
+	/* Validate location method priority Kconfig's */
+	enum location_method method_first =
+		location_str_to_method(CONFIG_LOCATION_REQUEST_DEFAULT_METHOD_FIRST);
+	enum location_method method_second =
+		location_str_to_method(CONFIG_LOCATION_REQUEST_DEFAULT_METHOD_SECOND);
+	enum location_method method_third =
+		location_str_to_method(CONFIG_LOCATION_REQUEST_DEFAULT_METHOD_THIRD);
+
+	__ASSERT(
+		method_first != LOCATION_METHOD_UNKNOWN,
+		"CONFIG_LOCATION_REQUEST_DEFAULT_METHOD_FIRST must be "
+		"'gnss, 'wifi' or 'cellular'");
```

Comment:
`CONFIG_LOCATION_REQUEST_DEFAULT_METHOD_FIRST` is a string now. There are older commits (which you can check from this PR) which use choice symbol.
The code is a bit more clean with strings I guess. However, the drawback is that we cannot use BUILD_ASSERT's with strings or I haven't found a way for that. Run-time assert's in library initialization (in practice in device boot) would be fine but they cost flash use. Currently 5 asserts are ~900 bytes of extra flash, which sounds a bit too much.

@simensrostad Any opinions? You can see the implementation with `choice` Kconfig in commit "lib: location: Kconfig for method priorities"

Reply:
I think I prefer the choice symbol approach. The `__ASSERT_NO_MSG()` flavor should decrease the flash size a bit, if you think its fine omitting the message argument.

Reply:
Yep, but getting into correct line requires some reverse engineering for a beginner user.

Reply:
We've discussed this and are going towards using choice symbols in Kconfig. This is mainly due to the memory requirements for assertions.